loopchain/blockchain/blockchain.py

```diff
@@ -269,25 +272,21 @@ def add_block(self,
 
     def __add_block(self, block: Block, confirm_info, need_to_write_tx_info=True, need_to_score_invoke=True):
         with self.__add_block_lock:
-            invoke_results = self.__invoke_results.get(block.header.hash.hex(), None)
-            if invoke_results is None and need_to_score_invoke:
-                if block.header.height == 0:
-                    block, invoke_results = ObjectManager().channel_service.genesis_invoke(block)
-                else:
-                    block, invoke_results = ObjectManager().channel_service.score_invoke(block, self.__last_block)
+            receipts, next_prep = self.__invoke_results.get(block.header.hash.hex(), (None, None))
+            if receipts is None and need_to_score_invoke:
+                self.get_invoke_func(block.header.height)(block, self.__last_block)
 
             try:
-                if need_to_write_tx_info:
-                    self.__add_tx_to_block_db(block, invoke_results)
+                if not need_to_write_tx_info:
```

Comment:
Is there any case that we don't need to write tx info anymore?

It was because of writing tx results before 'score write commit'.
Above logic is removed now. So this flag has become useless.

Reply:
It is a kind of trick to keep the program logic the same as before. It is not a modification of the flow here.

Reply:
예전 코드는 need_to_write_tx_info 가 있을때에만 add_tx_to_block_db 를 수행하도록 되어 있었기에 같은 흐름을 만들기 위해서 write_block_data 에 receipts 를 None 으로 전달하게 하여 동일하게 동작하게 한 것입니다. 이 곳의 코드 로직을 개선한 것은 아닙니다. 이 PR은 논리적인 리팩토링을 한것이 아니라 전적으로 같은 흐름을 유지하면서 invoke 함수들의 위치만 channel_service 에서 blockchain 으로 옮겨왔습니다. 이유는 invoke 할때 생성되는 정보를 blockchain 이 __invoke_result 로 관리를 하고 있는데 여기에 invoke 후 복수의 값을 저장할 필요가 생겼기 때문입니다.